includes/classes/shopping_cart.php

```diff
@@ -1071,75 +1133,68 @@ public function attributes_price($product_id)
     /**
      * Calculate one-time price of attributes for a given item
      *
-     * @param mixed $product_id the product ID of the item to check
+     * @param mixed $uprid the product ID of the item to check
      * @param float $qty item quantity
      * @return float the price of the items attributes
      */
-    public function attributes_price_onetime_charges($product_id, $qty)
+    public function attributes_price_onetime_charges($uprid, $qty)
     {
         global $db;
 
         $attributes_price_onetime = 0;
```

Comment:
For consistency with the other refactorings to early-return, this line 1144 `$attributes_price_onetime = 0;` should be moved below the early-return on line 1149-1150.